Content.Server/_DV/CosmicCult/CosmicGlyphSystem.cs

```diff
@@ -0,0 +1,132 @@
+    }
+
+    #region Base trigger
+
+    private void OnExamine(Entity<CosmicGlyphComponent> uid, ref ExaminedEvent args)
+    {
+        if (HasComp<CosmicCultComponent>(args.Examiner))
+        {
+            args.PushMarkup(Loc.GetString("cosmic-examine-glyph-cultcount", ("COUNT", uid.Comp.RequiredCultists)));
+            switch (uid.Comp.GlyphName) // This seems like the most straightforward way to do this, rather than making seperate examine events for each glyph component
+            {
+                case "knowledge":
+                    args.PushMarkup(Loc.GetString("cosmic-examine-glyph-knowledge"));
+                    break;
+                case "truth":
+                    args.PushMarkup(Loc.GetString("cosmic-examine-glyph-truth"));
+                    break;
+                case "cessation":
+                    args.PushMarkup(Loc.GetString("cosmic-examine-glyph-cessation"));
+                    break;
+                case "blades":
+                    args.PushMarkup(Loc.GetString("cosmic-examine-glyph-blades"));
+                    break;
+                case "warding":
+                    args.PushMarkup(Loc.GetString("cosmic-examine-glyph-warding"));
+                    break;
+                case "projection":
+                    args.PushMarkup(Loc.GetString("cosmic-examine-glyph-projection"));
+                    break;
+            }
+        }
+        else
+            args.PushMarkup(Loc.GetString("cosmic-examine-text-glyphs"));
+    }
+
+    private void OnUseGlyph(Entity<CosmicGlyphComponent> uid, ref ActivateInWorldEvent args)
+    {
+        Log.Debug($"Glyph event triggered!");
+        var tgtpos = Transform(uid).Coordinates;
+        var userCoords = Transform(args.User).Coordinates;
+        if (args.Handled || !userCoords.TryDistance(EntityManager, tgtpos, out var distance) || distance > uid.Comp.ActivationRange || !HasComp<CosmicCultComponent>(args.User))
+            return;
+        var cultists = GatherCultists(uid, uid.Comp.ActivationRange);
+        if (cultists.Count < uid.Comp.RequiredCultists)
+        {
+            _popup.PopupEntity(Loc.GetString("cult-glyph-not-enough-cultists"), uid, args.User);
+            return;
+        }
+
+        args.Handled = true;
+        var damageSpecifier = new DamageSpecifier();
+        var tryInvokeEv = new TryActivateGlyphEvent(args.User, cultists);
+        RaiseLocalEvent(uid, tryInvokeEv);
+        if (tryInvokeEv.Cancelled)
+            return;
+
+        damageSpecifier.DamageDict.Add("Asphyxiation", uid.Comp.ActivationDamage / cultists.Count);
+        foreach (var cultist in cultists)
+        {
+            DealDamage(cultist, damageSpecifier);
+        }
+
+        _audio.PlayPvs(uid.Comp.GylphSFX, tgtpos, AudioParams.Default.WithVolume(+1f));
+        Spawn(uid.Comp.GylphVFX, tgtpos);
+        QueueDel(uid);
+    }
+    #endregion
+
+    #region Housekeeping
+    private void DealDamage(EntityUid user, DamageSpecifier? damage = null)
```

Comment:
There's no reason for this to have a default parameter
```suggestion
    private void DealDamage(EntityUid user, DamageSpecifier? damage)
```